pennylane/devices/experimental/default_qubit_2.py

```diff
@@ -283,6 +284,270 @@ def compute_derivatives(
             f"{self.name} cannot compute derivatives via {execution_config.gradient_method}"
         )
 
+    def execute_and_compute_derivatives(
+        self,
+        circuits: QuantumTape_or_Batch,
+        execution_config: ExecutionConfig = DefaultExecutionConfig,
+    ):
+        is_single_circuit = False
+        if isinstance(circuits, QuantumScript):
+            is_single_circuit = True
+            circuits = [circuits]
+
+        if self.tracker.active:
+            for c in circuits:
+                self.tracker.update(resources=c.specs["resources"])
+            self.tracker.update(batches=1, executions=len(circuits))
+            self.tracker.update(derivative_batches=1, derivatives=len(circuits))
```

Comment:
```suggestion
            self.tracker.update(batches=1, executions=len(circuits), execute_and_derivative_batches=1, derivatives=len(circuits))
```
We can update everything at once.

I've often added this on my private branches, but never got around actually to make the change.  It would be nice to track whether or not `execute_and_compute_derivatives` is called instead of `execute` followed by `compute_derivatives`.  The names may need some assistance.